### ISSUE TITLE
Exporting variables with : type inference causes get_property_list() to return an incorrect "usage"

### ISSUE THREAD

**Issue by rambda** [godotengine/godot#59157]
### Godot version

3.5.beta1

### System information

Windows 10

### Issue description

For variables that declared like these:
```
export var no_type = ""
export var no_type_with_value = ""
export(String) var type_hint = ""
export(String) var type_hint_with_value = ""
export(String) var type_hint_plus_infer: String
export(String) var type_hint_plus_infer_with_value := ""
export var type_infer_resource: Resource
export var type_infer_with_value := ""
```
`get_property_list()` will return 
`{class_name:, hint:0, hint_string:, name:s, type:4, usage:8199}`
8199 == PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_SCRIPT_VARIABLE witch is correct.

___
In contrast, when
```
export var type_infer_string: String
export var type_infer_array: Array
export var type_infer_node_path: NodePath
```
`get_property_list()` will return 

`{class_name:, hint:0, hint_string:, name:s, type:4, usage:7}`

7 == PROPERTY_USAGE_DEFAULT, which is the same as an non-exported variable. This bug causes problems for tool scripts / plugins.
___
```
@export var no_type = ""
@export var no_type_with_value = ""
@export var type_infer_string: String
@export var type_infer_array: Array
@export var type_infer_node_path: NodePath
@export var type_infer_resource: Resource
@export var type_infer_with_value := ""
```
All returns `usage:8199` fine in 4.0.apha4.

### Steps to reproduce

Yes.

### Minimal reproduction project

[ExportVarUsage.zip](https://github.com/godotengine/godot/files/8250187/ExportVarUsage.zip)

**Comment by AnidemDex** [comment]
> 7 == PROPERTY_USAGE_DEFAULT, which is the same as an non-exported variable. This bug causes problems for tool scripts / plugins.

Just wanted to say that `PROPERTY_USAGE_DEFAULT` is `PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR | PROPERTY_USAGE_NETWORK`, so it'll be saved _and_ exposed in editor (as if it were an exported variable). Anyway, I can agree that is really important that exported variables should be marked with `PROPERTY_USAGE_SCRIPT_VARIABLE`.

I used [gd.tumeo.space](https://gd.tumeo.space/) (wich uses Godot 3.2.2) and [gdscript online](https://gdscript-online.github.io/) (wich uses Godot 3.4.2) with the following script, and I can confirm this issue in both playgrounds:
```gdscript
extends Node

export var no_type = ""
export var no_type_with_value = ""
export(String) var type_hint = ""
export(String) var type_hint_with_value = ""
export(String) var type_hint_plus_infer: String
export(String) var type_hint_plus_infer_with_value := ""
export var type_infer_resource: Resource
export var type_infer_with_value := ""

#export var type_infer_string: String
#export var type_infer_array: Array
#export var type_infer_node_path: NodePath

	
func _ready():
	print("--- Properties marked with SCRIPT_VARIABLE ---")
	for property in get_property_list():
		if bool(property.usage & PROPERTY_USAGE_SCRIPT_VARIABLE):
			print(property)
	print("\n")
	print("--- Properties marked with DEFAULT ---")
	for property in get_property_list():
		if bool(property.usage & PROPERTY_USAGE_DEFAULT):
			print(property)
```

**Comment by akien-mga** [comment]
Fixed by #60626.
